sample-xml/app/src/main/java/co/nimblehq/sample/xml/ui/screens/home/HomeViewModel.kt

```diff
@@ -32,10 +40,31 @@ class HomeViewModel @Inject constructor(
                     _uiModels.emit(uiModels)
                 }
             hideLoading()
+
+            requestPermissions()
         }
     }
 
     fun navigateToSecond(uiModel: UiModel) {
         execute { _navigator.emit(NavigationEvent.Second(uiModel)) }
     }
+
+    private suspend fun requestPermissions() {
+        _requestPermissions.emit(
+            arrayOf(
+                Manifest.permission.ACCESS_FINE_LOCATION,
+                Manifest.permission.CAMERA
+            )
+        )
+    }
+
+    fun onPermissionResult(permissionResult: PermissionResult) {
+        when (permissionResult) {
+            is Granted -> Timber.d("${permissionResult.grantedPermissions} granted")
+            is Denied.JustDenied -> Timber.d("${permissionResult.deniedPermissions} denied")
+            is Denied.NeedsRationale -> Timber.d("${permissionResult.deniedPermissions} needs rationale")
+            is Denied.DeniedPermanently -> Timber.d("${permissionResult.deniedPermissions} denied for good")
+            is Cancelled -> Timber.d("request cancelled")
+        }
+    }
```

Comment:
I personally think permission request is a UI action, so the logic to request permission should be in the View layer 💭 What do you think? What's the benefit of keeping it in VM layer?

Reply:
`onPermissionResult` could be in VM layer 💭